### DISCUSSION TITLE
showing subject ID

### DISCUSSION THREAD

**Discussion by veranicolette** [jspsych/jsPsych#1188] [Q&A] [answered]
In my experiment, I am generating a random ID for each participant. I was wondering how I could print this out at the end for the participant to copy? I had an idea on how to do it, but it's not working.
`var code = {
            
type: "html-keyboard-response",
            choices: jsPsych.NO_KEYS,
            stimulus: "You have completed the main task! Please copy the following code to the MTurk page. You can then close this window.  <br>" + jsPsych.data.get().subject_id + "<br> Your browser may ask you to confirm that you want to to leave this page, just say leave, we have recorded your performance."
        }
        timeline.push(code);`

**Comment by becky-gilbert** [comment, marked as answer]
Hi @veranicolette, the reason that this isn't working is because, to generate the stimulus dynamically, you need to turn the value of the [stimulus parameter into a function](https://www.jspsych.org/overview/trial/#dynamic-parameters) that returns the string.

Also, to access the value of "subject_id" in the data, I think you need to first get the values out of the jsPsych data collection like this: `jsPsych.data.get().values()`. This will return an array of the data for all trials, so you then need to index this array to get the data from a single trial. For example, adding the 0 index will give you the data from the first trial in the experiment: `jsPsych.data.get().values()[0]`. Once you have the data from a single trial, you can access the variable names/values that are stored in the trial data, like "subject_id".

So if "subject_id" is stored in the first trial data, you could do this:
```javascript
var code = {
  type: "html-keyboard-response",
  choices: jsPsych.NO_KEYS,
  stimulus: function() {
    return "You have completed the main task! Please copy the following code to the MTurk page. You can then close this window. " + jsPsych.data.get().values()[0].subject_id + ". Your browser may ask you to confirm that you want to to leave this page, just say leave, we have recorded your performance."
  }
}
timeline.push(code)
```
